integrations/ollama/src/ollama_haystack/chat/chat_generator.py

```diff
@@ -0,0 +1,97 @@
+from typing import Any, Dict, List, Optional
+
+import requests
+from haystack import component
+from haystack.dataclasses import ChatMessage
+from requests import Response
+
+
+@component
+class OllamaChatGenerator:
+    """
+    Chat Generator based on Ollama. Ollama is a library for easily running LLMs locally.
+    This component provides an interface to generate text using a LLM running in Ollama.
+    """
+
+    def __init__(
+        self,
+        model: str = "orca-mini",
+        url: str = "http://localhost:11434/api/chat",
+        generation_kwargs: Optional[Dict[str, Any]] = None,
+        template: Optional[str] = None,
+        timeout: int = 30,
+    ):
+        self.timeout = timeout
+        self.template = template
+        self.generation_kwargs = generation_kwargs or {}
+        self.url = url
+        self.model = model
+
+    def _get_telemetry_data(self) -> Dict[str, str]:
+        """
+        Data that is sent to Posthog for usage analytics.
+        """
+        return {"model": self.model}
+
+    def _message_to_dict(self, message: ChatMessage) -> Dict[str, str]:
+        return {"role": message.role.value, "content": message.content}
+
+    def _chat_history_to_dict(self, messages: List[ChatMessage]) -> List[Dict[str, str]]:
+        """
+        Returns A dictionary of JSON arguments for a POST request to an Ollama service
+        :param messages: A list of chat messages
+        :return: A list of chat messages as dictionaries
+        """
+        return [self._message_to_dict(message) for message in messages]
+
+    def _create_json_payload(self, messages: List[ChatMessage], generation_kwargs=None) -> Dict[str, Any]:
+        """
+        Returns A dictionary of JSON arguments for a POST request to an Ollama service
+        :param messages: A history/list of chat messages
+        :param generation_kwargs:
+        :return: A dictionary of arguments for a POST request to an Ollama service
+        """
+        generation_kwargs = generation_kwargs or {}
+        return {
+            "messages": self._chat_history_to_dict(messages),
```

Comment:
```suggestion
            "messages": [self._message_to_dict(message) for message in messages],
```

this can be done here